src/main/java/com/google/sps/Matrix.java

```diff
@@ -0,0 +1,172 @@
+enum Cell {
+    /*
+    0 --> dead
+    1 --> alive
+    2 --> alive with mask
+    3 --> alive with vaccine
+    4 --> alive with mask and vaccine
+    5 --> infected
+    6 --> infected with mask
+    7 --> infected with vaccine
+    8 --> infected with mask and vaccine
+    */
+
+    DEAD ((byte) '0'),
+    ALIVE ((byte) '1');
+  
+    private byte cellCode;
+  
+    public byte getCode() {
+        return this.cellCode;
+    }
+  
+    private Cell(byte cellCode) {
+        this.cellCode = cellCode;
+    }
+}
+
+public class Matrix {
+
+    private byte[][] matrix;
+
+    public Matrix(String stringMatrix) {
+        stringToMatrix(stringMatrix);
+    }
+
+    public void updateMatrix() {
+        int rows = this.matrix.length,
+            columns = this.matrix[0].length;
+
+
+        byte[][] tempMatrix = getMatrixCopy(this.matrix);
+
+        for (int row = 0; row < rows; row++) {
+            for (int column = 0; column < columns; column++) {
+                this.updateCell(row, column, tempMatrix);
+            }
+        }
+
+        System.out.println();
+        printMatrix(this.matrix);
+    }
+
+    private void updateCell(int row, int column, byte[][] tempMatrix) {
+        int neighbors = getNumberOfNeighbors(row, column, tempMatrix);
+
+        byte currentCellCode = tempMatrix[row][column];
+
+        if (currentCellCode == Cell.ALIVE.getCode()) {
+            if (neighbors < 2 || neighbors > 3) {
+                this.matrix[row][column] = Cell.DEAD.getCode();
+            }
+        }
+        else if (currentCellCode == Cell.DEAD.getCode()) {
+            if (neighbors == 3) {
+                this.matrix[row][column] = Cell.ALIVE.getCode();
+            }
+        }
+
+    }
+
+    private int getNumberOfNeighbors(int row, int column, byte[][] tempMatrix) {
+        int neighbors = 0;
+
+        neighbors += this.hasAliveNeighbor(row-1,column-1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row-1,column,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row-1,column+1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row,column+1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row+1,column+1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row+1,column,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row+1,column-1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row,column-1,tempMatrix) ? 1 : 0;
```

Comment:
This could be expressed as a loop w/in a loop. As long as you're only looking @ 8 neighbors, the amount of code is probably roughly the same. Otoh, if you ever wanted to look and neighbors an additional level out, this approach would be much harder to adapt to that. No need to change for now, just something to think about.

Reply:
Oh yes, I already have a TODO to implement a for loop :)